aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -218,13 +221,32 @@ def __init__(
         cors: bool,
         compress: bool,
         cache_control: Optional[str],
+        middleware: Optional[List[Callable]],
     ):
         self.method = method.upper()
         self.rule = rule
         self.func = func
         self.cors = cors
         self.compress = compress
         self.cache_control = cache_control
+        self.middleware = middleware or []
+
+    def __call__(self, router_middleware: List[Callable], app, args: Dict[str, str]):
+        """Builds the middleware stack using global and route middlewares, redefining the original handler function"""
+        if not self._middleware_built:
+            # prepend global router middleware first
+            all_middleware = list(router_middleware) + list(self.middleware)
+
+            # IMPORTANT: # this must be the last mdidleware in the stack to avoid breaking changes
+            # for the registered API call signature (Maintain Backward Compatibility)
+            all_middleware.append(registered_api_middleware)
```

Comment:
I'm not entirely sure i understood this part - is that due to `app(event, context)` vs `(app.resolve(event, context)`?

We should add more context in here as I'll surely forget.

Reply:
Yeah - this is for complete backward compatibility and retaining the existing call signature of route functions themselves.  Without this route function signatures would need to match the middleware signature of `myfunc(app: ApiGatewayResolver, **kwargs)` which is how each middleware in the chain calls `get_response` or `next_handler`.

Lets talk more and get the context exactly right - this is the only fuzzy piece and was expecting to do more here.

Reply:
Can possibly view this as "tech-debt" item to remove if we decide to change the API route signature to include "app: ApigateWayResolver" as the first argument - which is a breaking change.